services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/seven_zip_wrapper.py

```diff
@@ -0,0 +1,61 @@
+import asyncio
+import logging
+import re
+from pathlib import Path
+
+from models_library.basic_types import IDStr
+from servicelib.progress_bar import ProgressBarData
+
+from ..core.errors import SevenZipError
+from ..core.utils import async_command
+
+_logger = logging.getLogger(__name__)
+
+
+async def _get_file_count(zip_path: Path) -> int:
+    result = await async_command(f"7z l {zip_path}")
+    if not result.success:
+        raise SevenZipError(command=result.command, command_result=result.message)
+
+    match = re.search(r"\s*(\d+)\s*files", result.message)
```

Comment:
you can create a variable that contains the compiled pattern for added performance, and also use named groups for added understanding. ```r"\s*(?P<file_count>\d+)\s*files"``` and then you can use ```match.group('file_count')```
what also is possible is to use the parse library that makes it a bit more understandable as well. https://github.com/r1chardj0n3s/parse
and question, after file_count is there something no space? why not use ```+``` ?